src/browser/chrome.ts

```diff
@@ -40,16 +40,54 @@ export const closeBrowser = async () => {
 };
 
 const WINDOW_VARIABLE_NAME = `___webrender`;
+const LAST_RESPONSE_AT_VARIABLE_NAME = `__webrenderLastResponseAt`;
+const COUNTER_REQUEST_UNFINISHED_VARIABLE_NAME = `__webrenderCounterReqUnfinished`;
 export const getPageInitScriptFor = (
   js: string,
-  jsOn: JsOn = "commit"
-) => `try {
-  const AsyncFunction = Object.getPrototypeOf(
+  jsOn: JsOn = "commit",
+  timeout: number
+) => {
+  const PAGE_TRUE_LOAD = `
+  async pageTrueLoad ({
+    idleTimeout = 2000, resolveBeforeRenderingTimeout = 2000
+  } = {}) {
+    const timeoutPromise = new Promise((resolve) => {
+      setTimeout(() => {
+        console.error('Page is not loaded in time');
+        resolve();
+      }, ${timeout} - resolveBeforeRenderingTimeout);
+    });
+
+    const loadPromise = new Promise((resolve) => {
+      addEventListener('load', () => {
+        while (true) {
+          const lastResponseAt = +localStorage.getItem(
+            '${LAST_RESPONSE_AT_VARIABLE_NAME}'
+            );
+          const counterRequestsUnfinished = +localStorage.getItem(
+            '${COUNTER_REQUEST_UNFINISHED_VARIABLE_NAME}'
+          );
+          const isLoaded = lastResponseAt && Date.now() - lastResponseAt > idleTimeout;
+          if (isLoaded && counterRequestsUnfinished === 0) {
+            resolve();
+            break;
+          } else {}
```

Comment:
A sync while loop will just block the thread forever, you need to add some kind of a timeout or await promise